src/dmd/root/array.d

```diff
@@ -591,3 +591,81 @@ unittest
     static assert (!is(typeof(Array!(OtherStruct).init.sort())));
     static assert (!is(typeof(Array!(OtherStruct).init.sort!pred)));
 }
+
+/**
+ * Iterates the given array and calls the given callable for each element.
+ *
+ * Use this instead of `foreach` when the array may expand during iteration.
+ *
+ * Params:
+ *  callable = the callable to call for each element
+ *  array = the array to iterate
+ */
+template each(alias callable, T)
+if (is(typeof(callable(T.init))))
+{
+    void each(ref Array!T array)
```

Comment:
Note sure if this overload is required outside of your tests.

Reply:
There are many cases where a plain `Array` instead of a pointer to an `Array` is used. Here's one example:

https://github.com/dlang/dmd/blob/187a7fb1bad33c5c067515e5c82b8eb3801c273e/src/dmd/dclass.d#L163

Some of the existing utility functions that accept `Array` takes it by `ref` instead of by pointer.